src/System.Runtime/tests/System/ExitCodeTests.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.Tests
+{
+    public class ExitCodeTests : RemoteExecutorTestBase
+    {
+        private const int SIGTERM = 15;
+
+        [DllImport("libc", SetLastError = true)]
+        private static extern int kill(int pid, int sig);
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData(0)]
+        [InlineData(42)]
+        [PlatformSpecific(TestPlatforms.AnyUnix)] // SIGTERM signal.
+        public void SigTermExitCode(int? exitCodeOnSigterm)
+        {
+            Action<string> action = (string sigTermExitCode) =>
+            {
+                if (!string.IsNullOrEmpty(sigTermExitCode))
+                {
+                    AppDomain.CurrentDomain.ProcessExit += (_1, _2) =>
```

Comment:
Nit: Instead of `(_1, _2) =>`, you could write `delegate { … };`, and avoid having to use throwaway names.  Either's fine.